paddle/fluid/framework/ir/fuse_optimizer_ops_pass/fuse_optimizer_op_pass.cc

```diff
@@ -137,54 +136,77 @@ void FuseOptimizerOpPass::ApplyImpl(ir::Graph *graph) const {
       auto &fused_grad = result.Get<details::FusedGrads>(details::kFusedGrads);
       PADDLE_ENFORCE_NE(fused_grad.size(), 0,
                         "The fused gradient should not be empty.");
-      PADDLE_ENFORCE_EQ(fused_grad.size(), 1,
-                        "Because the dtype of those gradients "
-                        "is not unified, so the number of fused gradients is "
-                        "more than one, but it is not supported currently.");
+      if (fused_grad.size() > 1) {
+        // Note(chenweihang): Because the dtype of those gradients is not
+        //   unified,so the number of fused gradients is more than one,
+        //   but it is not supported currently.
+        return;
+      }
       auto &fused_vars = result.Get<details::FusedVars>(details::kFusedVars);
       auto iter =
           std::find(fused_vars.begin(), fused_vars.end(), fused_grad.front());
       PADDLE_ENFORCE_EQ(iter != fused_vars.end(), true,
-                        "Not find the fused_grad.");
+                        "Not found the fused gradient variable.");
       fused_vars_name[kGrad] = fused_grad.front();
 
       // Sort the parameters and auxiliary variables according
       // to parameters' name to make variables' name correspond correctly.
-      SortParametersAndAuxVars(params_and_dense_grads, &aux_var_set,
+      SortParametersAndAuxVars(params_and_dense_grads, &aux_var_map,
                                &opt_nodes);
       grad_fused = true;
     } else {
       VLOG(6) << "The number of new gradients is " << new_grad_idx.size();
       if (new_grad_idx.size() == 1) return;
       // NOTE(zcd): If the gradients of backward stage and optimization stage
       // have diff, Only take care of the the gradient of optimization stage.
-      GradientsFilter(new_grad_idx, &opt_nodes, &aux_var_set);
+      GradientsFilter(new_grad_idx, &opt_nodes, &aux_var_map);
     }
   }
 
-  // Check dtype
-  auto dtype = GetDtypeOfVar(vars_info, aux_var_set.at(kParam).front());
-  for (auto vars : aux_var_set) {
+  // Pass pre-condition check: check dtype of fusing vars
+  auto fusing_var_dtype =
+      GetDtypeOfVar(vars_info, aux_var_map.at(kParam).front());
+  for (auto vars : aux_var_map) {
     for (auto &var_name : vars.second) {
-      PADDLE_ENFORCE_EQ(dtype, GetDtypeOfVar(vars_info, var_name));
+      if (fusing_var_dtype != GetDtypeOfVar(vars_info, var_name)) {
+        // Note(chenweihang): Currently the fuse_optimizer_ops strategy
+        //   in mixed precision scenarios is not yet supported.
+        return;
+      }
+    }
+  }
+
+  // Pass pre-condition check: gradients generated op kernel
+  auto fusing_grad_var_names = aux_var_map.at(kGrad);
+  for (auto grad_var_name : fusing_grad_var_names) {
+    if (!GradGeneratedOpSupportGPU(vars_info, grad_var_name)) {
```

Comment:
Seems that it makes this pass does not support CPU case.

Reply:
done, check whether op with kernel both support CPU and GPU